### PR TITLE
adds coverage reporting and badges

### PR DESCRIPTION
**Pull request by bcoe**
This pull adds coverage and build status badges. Coverage is facilitated by [nyc](https://www.npmjs.com/package/nyc) and the hosted coverage service coveralls.io.

* run `npm run coverage` to get a human readable coverage report.
* run `npm run coverage -- --reporter=lcov` to get an HTML report over coverage in the /coverage folder.
* run `npm run coveralls` to report coverage to the [coveralls.io](https://coveralls.io/).
  * you'll need to setup your repo on coveralls.io and grap the `COVERALLS_REPO_TOKEN`.
  * on travis-ci.org, you'll need to set an environment variable with the value of `COVERALLS_REPO_TOKEN `

### CODE REVIEW

**Comment by jdalton** [comment]
Hey @bcoe! What's the benefit of using `nyc` over `istanbul` directly?

**Comment by bcoe** [comment]
@jdalton as you point out, nyc is just a thin wrapper for istanbul. I see two main benefits:

@isaacs and I have put a lot of work into making sure that nyc works great with subprocesses. While this allows it to instrument a larger variety of testing frameworks, e.g., `node-tap`, it has the added benefit that instrumentation becomes consistent and simple across testing frameworks. 

To instrument underscore with istanbul, you'd do something like this:

`istanbul cover qunit-cli test/*.js`, this instrumentation invocation varies widely depending on chosen framework.

Because of our handling of shebang, instrumentation simply becomes:

`nyc npm run test-node`

the other main advantage I see is that nyc makes it easier to spit out a text-lcov report which can be pushed directly to coveralls.io:

` nyc report --reporter=text-lcov`

This is slightly more of a hassle with `istanbul cover` (this would be a great candidate for a patch to istanbul itself, now that the text-lcov report exists).

File: .travis.yml [hunk 1]
```diff
@@ -29,4 +29,5 @@ env:
   global:
   - secure: bDZSBQfqr21hCayjcZ20IxrV6+XGhxQPFIfwWqEKLrF93Gu8LLVjZRxXE/mE8I8N4Z5WtDNb4ZHrm/TTzmcPa5MuHgIxEdknQCncobH8oimwc83SHwEPk6okeNKl39VlCjvvnmoe/V/KpnknuYn3Rqghtl/Uv9KLpCwskwjTtcw=
   - secure: SRECgXuwcZTcD3GVxTS2bYNgRyye4vq6BLrV2PH9FyNenowsKQR2EwlC/dppc1Q8NWMgv79J/R96q9JOFh+mEH9L5dlBb2yhnGH8amVeM/ChAJHT/F8YktKM453uVpz5fR00QcCQDDUOx6Pvx374ID0OKNpWKAkQBWA9mPTsLnE=
-  matrix: BROWSER=false
\ No newline at end of file
+  matrix: BROWSER=false
+after_success: npm run coveralls
```
Comment (megawac) [inline .travis.yml]: Do we need a token?

**Comment by megawac** [comment]
LGTM, mind rebasing?

**Comment by akre54** [comment]
We're typically against badges on the readme page. Mind pulling those out? I have no opinion about nyc vs istanbul, but +1 to better test coverage.

**Comment by megawac** [comment]
Thanks!

**Pull request closed without merging**